packages/backend-function/API.md

```diff
@@ -4,37 +4,20 @@
 
 ```ts
 
-import { AmplifyFunctionProps } from '@aws-amplify/function-construct-alpha';
-import { AmplifyLambdaFunction } from '@aws-amplify/function-construct-alpha';
+import { Construct } from 'constructs';
 import { ConstructFactory } from '@aws-amplify/plugin-types';
-import { ConstructFactoryGetInstanceProps } from '@aws-amplify/plugin-types';
+import { FunctionResources } from '@aws-amplify/plugin-types';
+import { ResourceProvider } from '@aws-amplify/plugin-types';
 
 // @public
-export class AmplifyFunctionFactory implements ConstructFactory<AmplifyLambdaFunction> {
-    static build: (props: AmplifyFunctionFactoryBuildProps) => Promise<AmplifyFunctionFactory>;
-    static fromDir: (props: AmplifyFunctionFactoryFromDirProps) => AmplifyFunctionFactory;
-    getInstance: ({ constructContainer, }: ConstructFactoryGetInstanceProps) => AmplifyLambdaFunction;
-}
+export const defineFunction: (props?: FunctionFactoryProps) => ConstructFactory<Construct & ResourceProvider<FunctionResources>>;
 
 // @public (undocumented)
-export type AmplifyFunctionFactoryBaseProps = {
-    name: string;
+export type FunctionFactoryProps = {

```

Comment:
I wonder if we should rename this type. data uses `defineData: (props: DataProps)`
auth uses `defineAuth: (props: AmplifyAuthProps)`

Reply:
Good call, will rename to `FunctionProps` (we should probably rename the auth props to `AuthProps` as well...